### PR TITLE
Allows user to edit events

### PR DESCRIPTION
**Pull request by farrahfazirah**
This change allows users to edit event details

Here are the screenshots below:

screen shot 2016-10-15 at 2 04 33 pm

The "edit" icon shows individually on each event

screen shot 2016-10-15 at 2 04 09 pm

However when you click on any "edit" icon, it shows up to edit ALL the events

### CODE REVIEW

**Comment by Drenmi** [comment]
Please rebase to latest `master`, so I can review the changes made by you. 😀 

1. Check out `master`: `git checkout master`
2. Pull latest `master`: `git pull origin master`
3. Run `bundle` and `rake db:drop db:create db:migrate db:seed`
4. Checkout your branch again: `git checkout feature/edit-events`
5. Rebase: `git rebase master`
6. Fix merge conflicts.
7. Push your branch: `git push -f origin feature/edit-events`

File: app/controllers/events_controller.rb [hunk 1]
```diff
@@ -28,6 +28,22 @@ def create
...
   end
 
+  def edit
+  end
+
+  def update
+    if @event.update(event_params)
+      redirect_to @events, notice: "Event was successfully updated."
+    else
+      render 'edit'
```
Comment (Drenmi) [inline app/controllers/events_controller.rb]: Please use double quotes throughout. 😀 This was also caught by RuboCop, so make sure to always rune `bundle exec rubocop`. 👮

File: app/controllers/events_controller.rb [hunk 2]
```diff
@@ -28,6 +28,22 @@ def create
...
+    end
+  end
+
+  def destroy
```
Comment (Drenmi) [inline app/controllers/events_controller.rb]: We don't allow deleting things from the application. 😀

File: app/views/clearance/sessions/_form.html.erb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  <div class="input-group">
+  <span class="input-group-addon" id="basic-addon1">Password</span>
+    <%= form.text_field :password %>
+  </div>
+
+  <div class="submit-field">
+    <%= form.submit %>
+  </div>
+
+<% end %>
```
Comment (Drenmi) [inline app/views/clearance/sessions/_form.html.erb]: Please configure your editor to insert a newline at the end of the file automatically. If you're using Sublime, the configuration option is named: `ensure_newline_at_eof_on_save`.

File: app/views/clearance/sessions/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<%= form_for :session, url: session_path do |form| %>
+  <div class="input-group">
+  <span class="input-group-addon" id="basic-addon1">E-mail</span>
+    <%= form.text_field :email %>
+  </div>
+
+  <div class="input-group">
+  <span class="input-group-addon" id="basic-addon1">Password</span>
```
Comment (Drenmi) [inline app/views/clearance/sessions/_form.html.erb]: Missing one level of indentation for the `span` element.

File: app/views/clearance/sessions/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<%= form_for :session, url: session_path do |form| %>
+  <div class="input-group">
+  <span class="input-group-addon" id="basic-addon1">E-mail</span>
```
Comment (Drenmi) [inline app/views/clearance/sessions/_form.html.erb]: Missing one level of indentation for the `span` element.

File: app/views/clearance/sessions/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<div id="clearance" class="sign-in">
```
Comment (Drenmi) [inline app/views/clearance/sessions/new.html.erb]: Not sure why this file is here. It doesn't look related to editing an event?

File: app/views/events/_edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,34 @@
+<div class="modal fade" id="donationeditModal" tabindex="-1" role="dialog" aria-labelledby="myModalLabel" aria-hidden="true">
+  <div class="modal-dialog modal-sm" role="document">
+    <div class="modal-content">
+      <div class="modal-body">
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+
+        <% @event.donations.each do |event| %>
```
Comment (Drenmi) [inline app/views/events/_edit.html.erb]: This opens a modal that has a form for every single donation. It should only have a form for the donation we clicked. 😀

File: app/views/layouts/application.html.erb [hunk 8]
```diff
@@ -7,8 +7,7 @@
     <%= csrf_meta_tags %>
   </head>
   <body>
-
-      <% if signed_in? %>
+    <% if signed_in? %>
```
Comment (Drenmi) [inline app/views/layouts/application.html.erb]: Thanks for fixing the small things that you find. 🌻

**Comment by Drenmi** [comment]
This is a good start, but I think there's a little bit of confusion on what needs to be done. This actually opens a form to edit the *donations* of an event, not the details of the event itself.

I think that might also be a user story, but they are two separate things. 😀 

Maybe we can put an "Edit event" button here?

screen shot 2016-10-15 at 15 35 59

WDYT?

File: app/controllers/events_controller.rb [hunk 9]
```diff
@@ -28,6 +28,22 @@ def create
     end
   end
 
+  def edit
```
Comment (Drenmi) [inline app/controllers/events_controller.rb]: Because we're editing from the event show page, we don't need this action, actually. 😀

File: app/views/clearance/sessions/_form.html.erb (hunk 3 above)
Comment (farrahfazirah) [inline app/views/clearance/sessions/_form.html.erb]: Sorry I actually meant to delete the views/clearance/sessions! Forgot that it was gonna be committed to this feature.

File: app/views/events/_edit.html.erb (hunk 7 above)
Comment (Drenmi) [inline app/views/events/_edit.html.erb]: Let's ✂️ this line, and ✂️ it's end somewhere below.

File: app/views/clearance/sessions/_form.html.erb (hunk 3 above)
Comment (Drenmi) [inline app/views/clearance/sessions/_form.html.erb]: Oooh. If this is not needed, I would be okay with removing it in this PR as a separate commit, actually. It's borderline on what is "housekeeping" and not. 😀
